src/binary/ibc-setup/commands/init.spec.ts

```diff
@@ -0,0 +1,89 @@
+import fs from 'fs';
+
+import test from 'ava';
+import axios from 'axios';
+import sinon from 'sinon';
+
+import { Options, run } from './init';
+
+const consoleLog = sinon.stub(console, 'log');
+const fsExistSync = sinon.stub(fs, 'existsSync');
+const fsMkdirSync = sinon.stub(fs, 'mkdirSync');
+const axiosGet = sinon.stub(axios, 'get');
+const fsReadFileSync = sinon.stub(fs, 'readFileSync');
+const fsWriteFileSync = sinon.stub(fs, 'writeFileSync');
+const fsLstatSync = sinon.stub(fs, 'lstatSync');
+
+test.beforeEach(() => {
+  sinon.reset();
+
+  fsLstatSync.returns({
+    ...new fs.Stats(),
+    isDirectory: () => true,
+    isFile: () => true,
+  });
+});
+
+test('read existing registry.yaml', async (t) => {
+  const options: Options = {
+    home: '/home/user',
+    src: 'AAA',
+    dest: 'BBB',
+  };
+
+  fsExistSync.onCall(0).returns(true).onCall(1).returns(true);
+
+  const registryPath = `${options.home}/registry.yaml`;
+  const registryYaml = `
+  version: 1
+  `;
+
+  axiosGet.resolves({
+    data: registryYaml,
+  });
+  fsReadFileSync.returns(registryYaml);
+  fsWriteFileSync.returns(undefined);
+
+  await run(options);
+
+  t.assert(fsExistSync.calledWithExactly(options.home));
+  t.assert(fsExistSync.calledWithExactly(registryPath));
+  t.assert(fsMkdirSync.notCalled);
+  t.assert(axiosGet.notCalled);
+  t.assert(fsReadFileSync.calledOnceWith(registryPath));
+  t.assert(fsWriteFileSync.notCalled);
+  t.assert(consoleLog.calledWith({ version: 1 }));
+});
+
+test('initialize home directory and pull registry.yaml from remote', async (t) => {
+  const options: Options = {
+    home: '/home/user',
+    src: 'AAA',
+    dest: 'BBB',
+  };
+
+  fsExistSync.onCall(0).returns(false).onCall(1).returns(false);
+  fsMkdirSync.returns(options.home);
+
+  const registryPath = `${options.home}/registry.yaml`;
+  const registryYaml = `
+  version: 1
+  `;
+
+  axiosGet.resolves({
+    data: registryYaml,
+  });
+  fsReadFileSync.returns(registryYaml);
+  fsWriteFileSync.returns(undefined);
+
+  await run(options);
+
+  t.assert(fsExistSync.calledWithExactly(options.home));
+  t.assert(fsExistSync.calledWithExactly(registryPath));
+  t.assert(fsMkdirSync.calledOnceWith(options.home));
+  t.assert(axiosGet.calledOnce);
+  t.assert(fsReadFileSync.calledOnceWith(registryPath));
+  t.assert(fsWriteFileSync.calledOnceWith(registryPath, registryYaml));
+  t.assert(consoleLog.calledWithMatch(new RegExp(`at ${options.home}`)));
+  t.assert(consoleLog.calledWith({ version: 1 }));
+});
```

Comment:
Would be nice to have tests for
- insufficient read/write permissions
- axios error
- bad yaml file

Reply:
Sure! Those tests are not complete yet, it just a showcase :)

Reply:
I think the goal here is to show how to test. I would be happy for one failure (eg. axios error) now, and let's get `init` merged. Much of the work here is to define a path, not ensure we have checked every box (yet).